src/components/activity-header/sequence-nav.tsx

```diff
@@ -42,8 +42,9 @@ export class SequenceNav extends React.PureComponent <IProps> {
     const { activities } = this.props;
     const indexedActivities = this.createIndexedActivitiesMap();
     const selectedActivity = indexedActivities.get(item);
-    const activityIndex = activities?.findIndex((activity) => activity === selectedActivity);
-    activityIndex && this.props.onActivityChange(activityIndex);
+    const activityIndex = activities?.findIndex((activity) => activity === selectedActivity) || 0;
+    activityIndex >= 0 && this.props.onActivityChange(activityIndex);
+    activityIndex >= 0 && setQueryValue("sequenceActivity", activityIndex + 1);
```

Comment:
I might be mis-reading this, but it doesn't looks like it is using the activityId if it exists.   I see there is some logic below that is is using the activityId if it is exists.  Maybe that can be extracted to a function and reused here.